### ISSUE TITLE
It should give an error when accessing environment variables with dot

### ISSUE THREAD

**Issue by ale5000-git** [koalaman/shellcheck#2342]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"):  0.7.2
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

test=1 # this is just to fool ShellCheck

echo "${test.abc}"
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
It should give an error because environment variables with dot aren't accessible directly.

**Comment by kurahaupo** [comment]
That syntax is valid for `ksh` to access map elements (nested namespaces):
```lang=ksh
t=( foo=bar )
echo "${t.foo}"
```
Output:
> `bar`

It should be a warning on any other shell.

**Comment by ale5000-git** [comment]
It isn't valid at least in `bash` and `bysybox ash`

**Comment by kurahaupo** [comment]
> It isn't valid at least in `bash` and `bysybox ash`

As I already said, it's invalid in anything except `ksh`.

Busybox's built-in shell is so broken that it would be an enormous separate effort to identify all the uses of standard posix features that don't work in it. (Personally I don't think that effort would be worth it.)

Although the busybox shell is sometimes referred to as `ash`, but that's stealing the name of the real `ash`, which is a separate shell that is mostly POSIX compliant, though still somewhat limited.

**Comment by ale5000-git** [comment]
For my use `bysybox ash` (or whatever is) is perfect, since it have also versions for Windows and Android and my scripts are really universal.
I just need to test them both under busybox and with ShellCheck using "sh" mode.
